resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -5,293 +5,1124 @@
            attributeFormDefault="unqualified"
            elementFormDefault="qualified">
 
-    <xs:simpleType name="stringType">
-        <xs:restriction base="xs:string"/>
-    </xs:simpleType>
+    <!--| MAIN DOCUMENT HIERARCHY |-->
+    <!--__ROOT__:
+        ├ dds      [0~1],
+        |   ├ profiles [0~1],
+        |   ├ log      [0~1],
+        |   └ types    [0~1], (rooted definition)
+        |
+        ├ profiles [0~1],
+        ├ log      [0~1],
+        └ types    [0~1] (standalone definition) -->
+    <!-- dds, profiles, log or types can be the ROOT element of the profile -->
+    <xs:element name="dds">
+        <xs:complexType>
+            <xs:all>
+                <xs:element name="profiles" type="profilesType" minOccurs="0" maxOccurs="1"/>
+                <xs:element name="types" type="typesType" minOccurs="0" maxOccurs="1"/>
+                <xs:element name="log" type="logType" minOccurs="0" maxOccurs="1"/>
+            </xs:all>
+        </xs:complexType>
+    </xs:element>
+    <xs:element name="profiles" type="profilesType"/>
+    <xs:element name="types" type="typesType"/>
+    <xs:element name="log" type="logType"/>
```

Comment:
I have tried with a stand-alone profile for the tag "log", and verification complains that element 'log' is not an element of schema.
From documentation here `https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/making_xml_profiles.html#rooted-vs-standalone-profiles-definition` it says that it could be defined this way though.

Reply:
Did you move the attribute `xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles"` from the DDS tag to the log tag?
He should also include in the documentation that the attribute must be in the first/parent element of the xml profile.

Reply:
You are right, this should be properly documented.

Reply:
It is in Creating an XML profiles file documentation section and Rooted VS Standalone section too.